examples/tfengine/app.hcl

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This example is still under development and is not considered stable!
+# Breaking changes to components for this example will not be part of releases.
+
+template "app" {
+  recipe_path = "../../templates/tfengine/recipes/app.hcl"
+  data = {
+    constants = {
+      shared = {
+        env_code        = "s"
```

Comment:
Is it described somewhere what this env_code is and what values it can take? I couldn't find a description of it in this PR.

Reply:
I will be adding a schema for this section later on but the main thing is that it should be a single letter usually and will be appended as a suffix to globally unique resources (such as projects and buckets).